src/server/publicApi.js

```diff
@@ -60,14 +56,37 @@ const runPublicApi = (gameServer) => {
     };
   });
 
-  router.get('/model/:id', async (ctx) => {
-    const matchID = ctx.params.id;
+  //authorise
+  router.use('/:matchID/', async (ctx, next) => {
```

Comment:
Do I understand it correctly that routes above this `router.use()` call are not authenticated and only the ones belows actually are?

If yes, I am not sure if this is a good solution. This will definitly lead to confusion if someone wants to add another endpoint in the future...  Would it be possible to attach the middleware only to certain routes (and make this explicit)?  Or alternatively, could you add something like a whitelist, such that the middleware by default wants to authorize everything and only the routes in the whitelist are not considered?

Reply:
It should be that the routes which match the path `'/:matchID/'` use the authorisation.
the syntax is `router.use(path, callback)` where the path is either a string or an array of paths.
It might be better to use `'/:matchID/*'` or to explicitly list all the complete paths: `['/:matchID/players', '/:matchID/model/', ... ]`

Reply:
Oh, I see. Cool!  I think `'/:matchID/*'` would be good. Then the logic is that everything specific to a game should be authenticated, which sounds reasonable to me.

Just for clearity, could you move the `router.use()` statement above all the route definitions? And could you please mention in the comment that the middleware only applies to the `'/:matchID/*'` routes, please?